controller/linodemachine_controller.go

```diff
@@ -278,14 +278,36 @@ func (r *LinodeMachineReconciler) reconcileCreate(
 	return ctrl.Result{}, nil
 }
 
+// createInstance provisions linode instance after checking if the request will be within the rate-limits
+// Note: it takes a lock before checking for the rate limits and releases it after making request to linode API or when returning from function
+func (r *LinodeMachineReconciler) createInstance(ctx context.Context, logger logr.Logger, machineScope *scope.MachineScope, createOpts *linodego.InstanceCreateOptions) (*linodego.Instance, error, bool) {
```

Comment:
nit: we don't need this to be a reconciler function. Might also want to move this into the helpers for the machine controller

Reply:
updated